Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.cpp

```diff
@@ -61,7 +62,7 @@ WakeLock& NavigatorScreenWakeLock::wakeLock(Navigator& navigator)
 WakeLock& NavigatorScreenWakeLock::wakeLock()
 {
     if (!m_wakeLock)
-        m_wakeLock = WakeLock::create();
+        m_wakeLock = WakeLock::create(downcast<Document>(m_navigator.scriptExecutionContext()));
```

Comment:
Why do we need to downcast this here?  It's just passed to the ContextDestructionObserver constructor.

Reply:
I am purposefully using tight typing (Document instead of ScriptExecutionContext) to make it clear that WakeLock is only exposed to Window contexts and not workers.

I am not trying to future proof it to make it work with workers because I don't know that there is such plan. In particular, the spec doesn't currently expose this to workers. Also, if somebody were to add support for workers, there would likely be a lot more work involved than just changing the type here.